### DISCUSSION TITLE
CRC ERROR but MD5SUM MATCHS (RESOLVED)

### DISCUSSION THREAD

**Discussion by mauruco** [sanni/cartreader#213] [Need help]
Hi guys, I started from scratch. The second attempt seems to be working.
My GameBoy game was successful.
My N64 games keep giving CRC error, but I took the dumps and for each MD5SUM I found an equivalent MD5SUM on google.
So I was in doubt, these MD5 below are good dumps or what's wrong now?

MD5 ae5107efdd3c210e1edd4acd9b3cac31  TUROK_DINOSAUR_.Z64
MD5 2a0a8acb61538235bc1094d297fb6556  ZELDAMAJORASMAS.Z64
MD5 70c525880240c1e838b8b1be35666c3b  GOLDENEYE.Z64
MD5 caf9a78db13ee00002ff63a3c0c5eabb  STARFOX64.Z64
MD5 3a67d9986f54eb282924fca4cd5f6dff  MARIOKART64.Z64
MD5 1add2c0217662b307cdfd876b35fbf7a  ResidentEvilII.Z64
MD5 c7f1a43764a26da2e43f2a36a5f76e4c  DUKENUKEM.Z64

**Comment by majorpbx** [comment]
Hello, First I would not use google for DAT matches. Use instead the website below as it's the only true trusted source of that data. Second, what version of the Sanni software are you using on your Arduino and is your SD Card setup correctly? If I were you I'd completely re-due those first following the wiki guides and then test again.

[https://datomatic.no-intro.org/](url)

**Reply by mauruco** [reply]
I finished assembling, I successfully dumped the SNES.
I found all md5 of my N64 dumps on datomatic, I tested all my SD cards, only N64 shows CRC error even though it was successful.
SD cards formatted exFAT, FAT32 (Windows & SD Formarter) and even then CRC error.
In short, GB works, SNES works, N64 works but CRC error.
It's great for me, thank you very much for your help.
At some point I will try more SD cards.

**Reply by PsyK0p4T** [reply]
Let's take Goldeneye (USA) for example, it has a lot of confirmations on Datomatic.

Your md5 is "70c525880240c1e838b8b1be35666c3b", and also on DoM.
Your crc32 should be "B6330846". Can you drag and drop your rom in a software to confirm this ? Like HashCalc ?
I checked again the n64.txt and the line is correct "b6330846,NGEE,12,5", so the crc detection should be fine. Please confirm the crc32 from the rom you generated with your reader.

**Reply by mauruco** [reply]
@PsyK0p4T 
GOLDENEYE.Z64                              b6330846
DUKENUKEM64.Z64                         dbfd5a53
GOLDENEYE.Z64                              b6330846
MARIOKART64.Z64                           434389c1
ResidentEvilII.Z64                               848fbc0d
STARFOX64.Z64                                b1fcaa9c
SUPERMARIO64.Z64                         3ce60709
TUROK_DINOSAUR_HUNTER.Z64   26c4f597
ZELDAMAJORASMASK.Z64              b428d8a7

![image](https://user-images.githubusercontent.com/11234529/127795106-fe450d79-a9fe-4706-b533-1b6f592a63e9.png)

**Reply by mauruco** [reply]
At the end of the dump process, a crc value appears, but it is definitely not the correct crc value. It's not the value of n64.txt, it's the value it uses to compare that is wrong. But if I take the card out and calculate the crc value on the PC, then it's the correct value.
This value the arduino saves in memory and uses to compare in the end?
Could it calculate from the file on the SD?

**Reply by sanni** [reply]
The current N64 code calculates the CRC from the memory of the Arduino because it's faster.
[Version 2.7](https://github.com/sanni/cartreader/releases/tag/V2.7) calculated the CRC from the file on the SD card.  You can try that older version and see if it makes a difference, it does need a FAT32 formatted SD card.

**Reply by mauruco** [reply]
![1627913596884](https://user-images.githubusercontent.com/11234529/127876551-71b42d48-0c7f-4151-9a2b-808e8e6576ae.jpg)
It worked @sanni  :)
Would it be possible to put a switch in the sketch "slowCRCCheck/fastCRCCheck"?
I would like to use the latest version.
Sorry if I'm asking for too much. I don't program C++ but I think I can help if you want.

**Reply by sanni** [reply]
Here is the latest V6.5 version but with the old N64 dumping and CRC code from V2.7. It still has the new SD library though. 

https://www.dropbox.com/s/qa8vjqhxpv8wz6l/V6.5_Portable_Test.zip?dl=1

**Reply by mauruco** [reply]
Thank you very much. I've already tested it with an N64 cartridge and it works perfectly (v6.5).

**Comment by sanni** [comment]
Ok I have uploaded this test version as V6.6 to the main github.

To enable slow crc calculation in option.h remove // in front of "#define slowcrc" like so:

// Define CRC method for dumping N64 ROMs, slow seems to be more compatible with some SD cards
#define slowcrc // crc will be calculated after dumping from SD card instead of during dumping from memory

**Reply by kdawg8** [reply]
Is V6.6 still available?  I don't see it on the main github page.  I'm having this issue on many (4 out of 9) Arduino modules and that would be great if there is a fix.  Thanks!

**Reply by sanni** [reply]
I have put together a V6.6 release just now.

You will have to remove the two // in front of #define slowcrc in line 41 of options.h before updating to enable slow crc.

**Reply by kdawg8** [reply]
Thanks sanni!  The slowcrc method fixed the issue on the Android modules which produced the N64 CRC error.
